res/controllers/Korg-KAOSS-DJ-scripts.js

```diff
@@ -0,0 +1,194 @@
+    var deck_playing = engine.getValue('[Channel' + deckNumber + ']', 'play_indicator');
+
+    // If in scratch mode or not playing enable vinyl-like control
+    if (deck.jogWheelsInScratchMode || !deck_playing ) {
+        if (value === ON) {
+            // Enable scratching on touch
+            engine.scratchEnable(deckNumber, 128, 33 + 1 / 3, alpha, beta);
+        } else if (value === OFF) {
+            // Disable scratching
+            engine.scratchDisable(deckNumber);
+        }
+    }
+};
+
+KAOSSDJ.wheelTurn = function(channel, control, value, status, group) {
+    var deck = KAOSSDJ.getDeckByChannel(channel);
+    var deckNumber = deck.deckNumber;
+    var newValue = 0;
+    if (value < 64) {
+        newValue = value;
+    } else {
+        newValue = value - 128;
+    }
+    if (engine.isScratching(deckNumber)) {
+        engine.scratchTick(deckNumber, newValue); 
+    } else {
+        engine.setValue('[Channel' + deckNumber + ']', 'jog', newValue); // Pitch bend
+    }
+};
+
+KAOSSDJ.wheelTurnShift = function(channel, control, value, status, group) {
+    var deck = KAOSSDJ.getDeckByChannel(channel);
+    if (deck.jogWheelsInScratchMode) {
+        // Fast scratch
+        var newValue = 0;
+        if (value < 64) {
+            newValue = value;
+        } else {
+            newValue = value - 128;
+        }
+        newValue = newValue * 4; // multiplier (to speed it up)
+        engine.scratchTick(deck.deckNumber, newValue); 
+    } else {
+        // Move beatgrid 
+        if(value === UP){
+            engine.setValue(deck.group, 'beats_translate_later', true);
+        } else if (value === DOWN){
+            engine.setValue(deck.group, 'beats_translate_earlier', true);
+        }
+    }
+};
+
+KAOSSDJ.scratchMode = function(channel, control, value, status, group) {
+    if (value === ON) {
+        // Turn scratch mode on jogwheel (LED is red)
+        KAOSSDJ.updateDeckByChannel(channel, 'jogWheelsInScratchMode', true);
+    } else if (value === OFF) {
+        // Turn scratch mode on jogwheel (LED is off)
+        KAOSSDJ.updateDeckByChannel(channel, 'jogWheelsInScratchMode', false);
+    }
+};
+
+KAOSSDJ.browseKnob = function(channel, control, value, status, group) {
+    var nval = (value > 0x40 ? value - 0x80 : value);
+    engine.setValue(group, "SelectTrackKnob", nval);
+};
+
+KAOSSDJ.leftFxSwitch = function(channel, control, value, status, group) {
+    var deck = KAOSSDJ.getDeckByChannel(channel);
+    if(value === ON) {
+        KAOSSDJ.updateDeckByChannel(channel, 'fx', true);
+    } else {
+        KAOSSDJ.updateDeckByChannel(channel, 'fx', false);
+    }
+};
+
+KAOSSDJ.rightFxSwitch = function(channel, control, value, status, group) {
+    if(value === ON) {
+        KAOSSDJ.updateDeckByChannel(channel, 'fx', true);
+    } else {
+        KAOSSDJ.updateDeckByChannel(channel, 'fx', false);
+    }
+};
+
+KAOSSDJ.controllerFxTouchMove = function(channel, control, value, status, group) {
```

Comment:
Does the controller send MIDI messages just for one axis of the touchpad? I would presume that it would send one MIDI message for the horizontal axis and one for the vertical axis so you could control two things at once. If it does that, you could have one axis control the dry/wet knob (`mix`) and the other control the superknob (`super1`).